### PR TITLE
Join Leave Driver API and minor updates to the existing NB APIs

### PR DESCRIPTION
**Pull request by mavenugo**

### CODE REVIEW

File: libnetwork_test.go [hunk 1]
```diff
@@ -160,18 +168,20 @@ func TestDuplicateNetwork(t *testing.T) {
...
 	if err != nil {
-		t.Fatal(err)
+		t.Fatal(err.Error())
 	}
 
-	_, err = controller.NewNetwork(netType, "testnetwork", "")
+	_, err = controller.NewNetwork(netType, "testnetwork", nil)
 	if err != nil {
-		t.Fatal(err)
+		t.Fatal("Expected to fail. But instead succeeded")
```
Comment (aboch) [inline libnetwork_test.go]: If you are checking for != nil then it was not expected to fail.

Comment (mavenugo) [inline libnetwork_test.go]: good catch. will fix it.

**Comment by aboch** [comment]
LGTM

**Comment by mrjana** [comment]
LGTM

**Pull request merged**